src/main/java/com/mindsnacks/zinc/classes/downloads/PriorityJobQueue.java

```diff
@@ -141,6 +141,13 @@ public void add(final Input element) {
         return (result != null) ? result : waitForFuture(element);
     }
 
+    public void reAdd(final Input element) throws JobNotFoundException {
+        checkServiceIsRunning(true, "Service should be running");
+        checkJobWasAlreadyAdded(element);
+        removeCachedFuture(element);
+        addElementToQueue(element);
```

Comment:
I'm concerned about this call... This operation doesn't use ```mLock``` but the ```get``` method doesn't either. Not sure if it's ok to add an element to the queue without locking it first :/